Library/OcAppleKernelLib/KextPatcher.c

```diff
@@ -84,33 +90,66 @@ PatcherInitContextFromBuffer (
   // and request PRELINK_KERNEL_IDENTIFIER.
   //
 
-  if (!MachoInitializeContext (&Context->MachContext, Buffer, BufferSize, 0)) {
+  if (!MachoInitializeContext (&Context->MachContext, Buffer, BufferSize, 0, Is32Bit)) {
     DEBUG ((DEBUG_INFO, "OCAK: Patcher init from buffer %p %u has unsupported mach-o\n", Buffer, BufferSize));
     return EFI_INVALID_PARAMETER;
   }
 
-  Segment = MachoGetSegmentByName64 (
-    &Context->MachContext,
-    "__TEXT"
-    );
-  if (Segment == NULL || Segment->VirtualAddress < Segment->FileOffset) {
-    return EFI_NOT_FOUND;
+  //
+  // 32-bit can be of type MH_OBJECT, which has all sections in a single unnamed segment.
+  // We'll fallback to that if there is no __TEXT segment.
+  //
+  if (Is32Bit) {
+    Segment32 = MachoGetSegmentByName32 (
```

Comment:
Do any 32-bit kexts have `__TEXT` segment? If not, maybe it is better to only use section lookup and/or fallback to `__TEXT` if it is less common?